openhtf/exe/__init__.py

```diff
@@ -87,17 +92,21 @@ def Start(self):
   def Stop(self):
     """Stop this test."""
     _LOG.info('Stopping test executor.')
-    if self._current_exit_stack:
-      # Tell the stack to exit.
-      with self._current_exit_stack.pop_all() as stack:
-        # Supress the error we're about to raise.
-        stack.push(lambda *exc_details: True)
-        raise TestStopError('Stopping.')
+    with self._lock:
+      if self._exit_stack:
+        # If self._exit_stack is not None, it will reset itself back to None
+        # when close()'d.
+        self._exit_stack.close()
```

Comment:
Hm, I wonder why I went with raising a suppressed error instead of this? I think it tells the callbacks something exceptional is happening (test stopping not finishing)?